tools/python_api/src_cpp/pyarrow/pyarrow_scan.cpp

```diff
@@ -30,7 +47,11 @@ static std::unique_ptr<function::TableFuncBindData> bindFunc(main::ClientContext
     }
     auto numRows = py::len(table);
     auto schema = Pyarrow::bind(table, returnTypes, names);
-
+    auto config = PyArrowScanConfig(input->config.options);
+    if (config.skipNum != 0 || config.limitNum != 0) {
+        // The following operation is zero copy as defined in pyarrow docs.
+        table = table.attr("slice")(config.skipNum, config.limitNum);
```

Comment:
Since skip and limit is processed and applied before bind data is even created, it's not necessary to include config in BindInput.